src/DacpacTool/Extensions.cs

```diff
@@ -107,26 +106,101 @@ public static void AddReference(this TSqlModel model, string referencePath, stri
 
             if (!string.IsNullOrWhiteSpace(externalParts))
             {
-                setMetadataMethod.Invoke(customData, new object[] { "ExternalParts", Identifier.EncodeIdentifier(externalParts) });
+                var parts = ParseExternalParts(externalParts);
+                if (!string.IsNullOrEmpty(parts))
+                {
+                    setMetadataMethod.Invoke(customData, new object[] {"ExternalParts", parts});
+                }
             }
 
             AddCustomData(dataSchemaModel, customData);
         }
 
+        public static string ParseExternalParts(string externalParts)
```

Comment:
I'm wondering if perhaps a regular expression would be a better solution for this parsing logic. It feels fairly complicated.